tailscale/resource_aws_external_id.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) David Bond, Tailscale Inc, & Contributors
+// SPDX-License-Identifier: MIT
+
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+func resourceAWSExternalID() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The aws_external_id resource allows you to mint an AWS External ID that Tailscale can use to assume an AWS IAM role that you create for the purposes of allowing Tailscale to stream logs to your S3 bucket. See the logstream_configuration resource for more.",
+		CreateContext: resourceAWSExternalIDCreate,
+
+		// No GET or DELETE endpoints in the API. This is a create-only resource.
+		ReadContext:   schema.NoopContext,
+		DeleteContext: schema.NoopContext,
+
+		Schema: map[string]*schema.Schema{
+			"external_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The actual External ID that Tailscale will supply when assuming your role. You probably want to reference this in your IAM role's trust policy.",
+			},
+			"tailscale_aws_account_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The AWS account from which Tailscale will assume your role. You probably want to reference this in your IAM role's trust policy.",
```

Comment:
Same here as above, I think we want stronger wording:
```suggestion
				Description: "The AWS account from which Tailscale will assume your role. You must reference this in your IAM role's trust policy. See https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html for more information on external IDs.",
```